### PR TITLE
Upgrade discord.js to v13

### PR DESCRIPTION
**Pull request by CodyJasonBennett**
Uses native bindings from the last major for interactions and largely depreciates usage of formerly private APIs.

### CODE REVIEW

File: src/utils/discord.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ export const validateMessage = message => {
     tts: Boolean(message.tts),
     flags: validateKeys(message.flags || message, INTERACTION_RESPONSE_FLAGS),
```
Comment (CodyJasonBennett) [inline src/utils/discord.js:115]: Might be able to rely on the API to sanitize flags for us:
 - https://github.com/discordjs/discord.js/blob/main/src/structures/MessagePayload.js#L153-L155

File: src/events/interactionCreate.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          if (!output) return;
+
+          const data = validateMessage(output);
+          await interaction.reply(data);
+          if (!data.components) return;
+
+          const message = await client.api
+            .webhooks(client.user.id, interaction.token)
+            .messages('@original')
+            .get();
```
Comment (CodyJasonBennett) [inline src/events/interactionCreate.js:32]: Would prefer if this was native or if there was a better way of keeping track of parent interactions/callbacks for buttons.

**Comment by CodyJasonBennett** [comment]
Hmm, it seems that the `engines` in Discord.js creeped up to v16 so our CI cannot install it. Might pose an issue for prod as well.

File: src/events/interactionCreate.js (hunk 2 above)
Comment (GmBodhi) [reply src/events/interactionCreate.js:32]: I think [`Interaction.reply`](https://discord.js.org/#/docs/main/stable/class/CommandInteraction?scrollTo=reply) will return something, if it doesn't then we can use [`Interaction.fetchReply`](https://discord.js.org/#/docs/main/stable/class/CommandInteraction?scrollTo=fetchReply)

Comment (GmBodhi) [reply src/events/interactionCreate.js:32]: Maybe, cache the command interaction and get the interaction when we get a button interaction

**Pull request merged**